Add support for custom HTML attributes to sidebar links (#774)
withastro/starlight: docs/src/content/docs/guides/sidebar.mdx

Reviewer: The example in docs is very pretty with its icon, but it may be slightly confusing as it’s kind of a “Draw the rest of the owl” example because we show “add class!” and then 🎉 magic icon appears, without showing how exactly that’s created.

Given the focus here is probably not to go in depth into different techniques for adding icons with CSS, I might suggest a simpler example e.g. with `style: 'color: red'` or something that shows the basic mechanism without needing to guess at some hidden code.

Author: I went for `font-style: italic` after trying for way too long to find a color matching the WCAG 2.0 level AA requirements for both the light & dark theme ^^ Hope it is obvious enough.

Reviewer: Looks good! We only need to demonstrate the concept here and this should do that nicely.

	]}
/>

## Custom HTML attributes

Links can also include an `attrs` property to add custom HTML attributes to the link element.

> An example use case could be to add a `target="_blank"` attribute to external links to open them in a new tab and also apply a custom `style` attribute to italicize the link label.

Reviewer: Suggesting a small rewording here inspired by recent guidance from @sarah11918 about introducing code examples:

```suggestion
In the following example, `attrs` is used to add a `target="_blank"` attribute, so that the link opens in a new tab, and to apply a custom `style` attribute to italicize the link label:
```
